[closed] Deprecate `toggleableActiveColor`

TahaTesser opened the pull request:
`toggleableActiveColor` is used in bunch of widgets where `colorScheme` is not adopted.
This PR deprecates `toggleableActiveColor` and updates all the widgets to use the colorScheme property.

- Deprecated `toggleableActiveColor` and added changes for dart fix
- Updated Switch, & SwitchListTile
- Updated Checkbox & CheckboxListTitle
- Updated Radio and RadioListTile
( had to expected colors in the test since Colors.blue[600] (default value of `toggleableActiveColor` is slightly different blue than `colorScheme.secondar`)

This is breaking as it would make `toggleableActiveColor` not have any effect on the above widgets

Related https://github.com/flutter/flutter/pull/90317 and #91772
Fixes https://github.com/flutter/flutter/issues/93709

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

TahaTesser committed:
Deprecate toggleableActiveColor and update switch

TahaTesser committed:
update switch_test

TahaTesser committed:
update checkbox

TahaTesser committed:
update CheckboxListTile

TahaTesser committed:
update Switchbox:ListTile

TahaTesser committed:
update radio

TahaTesser committed:
update radio theme doc

TahaTesser committed:
update RadioListTile

skia-gold commented:
Gold has detected about 1 new digest(s) on patchset 9.
View them at https://flutter-gold.skia.org/cl/github/95870

guidezpl commented on packages/flutter/lib/src/material/theme_data.dart:
```diff
@@ -799,6 +802,11 @@ class ThemeData with Diagnosticable {
     required this.primaryColorBrightness,
+    @Deprecated(
+      'No longer used by the framework, please remove any reference to it. '
+      'This feature was deprecated after v[TBD]',
```
TODO: add link to migration guide

guidezpl commented on packages/flutter/lib/src/material/theme_data.dart:
```diff
@@ -385,6 +384,11 @@ class ThemeData with Diagnosticable {
     Brightness? primaryColorBrightness,
+    @Deprecated(
+      'No longer used by the framework, please remove any reference to it. '
+      'This feature was deprecated after v[TBD]',
```
TODO: add link to migration guide

guidezpl commented on packages/flutter/lib/src/material/switch_list_tile.dart:
```diff
@@ -238,7 +238,7 @@ class SwitchListTile extends StatelessWidget {
   /// The color to use on the track when this switch is on.
   ///
-  /// Defaults to [ThemeData.toggleableActiveColor] with the opacity set at 50%.
+  /// Defaults to [ThemeData.colorScheme.secondary] with the opacity set at 50%.
```
I'm not sure that these doc references work. Can simplify to `[ColorScheme.secondary]`

guidezpl commented on packages/flutter/lib/fix_data.yaml:
```diff
@@ -147,6 +147,39 @@ transforms:
       - kind: 'removeParameter'
         name: 'primaryColorBrightness'
 
+  # Changes made in TBD
```
nit: please move to the top

guidezpl commented on packages/flutter/lib/src/material/switch_list_tile.dart:
```diff
@@ -238,7 +238,7 @@ class SwitchListTile extends StatelessWidget {
 
```
Please update the docs for the three instances of `activeColor`.

guidezpl commented on packages/flutter/lib/src/material/theme_data.dart:
```diff
@@ -1694,6 +1705,11 @@ class ThemeData with Diagnosticable {
     Brightness? primaryColorBrightness,
+    @Deprecated(
+      'No longer used by the framework, please remove any reference to it. '
+      'This feature was deprecated after v[TBD]',
```
TODO: add link to migration guide

guidezpl reviewed (changes requested):
Thanks for doing this! 

I believe this requires a (short) migration guide: https://github.com/flutter/flutter/wiki/Tree-hygiene#3-prepare-your-change.

Can you figure out which version to use for the deprecation? The wiki indicates the dev channel but now that it's deprecated, I'm not sure whether master or beta is more appropriate

TahaTesser committed:
Remove trailing space

skia-gold commented:
Gold has detected about 3 new digest(s) on patchset 10.
View them at https://flutter-gold.skia.org/cl/github/95870

TahaTesser committed:
Update ColorScheme doc references

skia-gold commented:
Gold has detected about 1 new digest(s) on patchset 11.
View them at https://flutter-gold.skia.org/cl/github/95870

TahaTesser committed:
nits changes

skia-gold commented:
Gold has detected about 3 new digest(s) on patchset 12.
View them at https://flutter-gold.skia.org/cl/github/95870

TahaTesser committed:
update doc and version

TahaTesser committed:
update doc

skia-gold commented:
Gold has detected about 5 new digest(s) on patchset 13.
View them at https://flutter-gold.skia.org/cl/github/95870

skia-gold commented:
Gold has detected about 2 new digest(s) on patchset 14.
View them at https://flutter-gold.skia.org/cl/github/95870

TahaTesser committed:
update fixes data

skia-gold commented:
Gold has detected about 3 new digest(s) on patchset 14.
View them at https://flutter-gold.skia.org/cl/github/95870

TahaTesser committed:
update deprecation notice

skia-gold commented:
Gold has detected about 1 new digest(s) on patchset 15.
View them at https://flutter-gold.skia.org/cl/github/95870

TahaTesser commented:
@guidezpl 
Resolved conflicsts, update branch and also updated deprecation to use beta release as in the [wiki](https://github.com/flutter/flutter/wiki/Tree-hygiene#deprecations) but still tests fails for deprecation notice 
Do you have any suggestions here?

guidezpl commented on packages/flutter/lib/fix_data.yaml:
```diff
@@ -14,6 +14,39 @@
+
+  # Changes made in https://github.com/flutter/flutter/pull/95870/
+  - title: "Remove 'toggleableActiveColor'"
+    date: 2022-01-03
```
nit: update date

guidezpl commented on packages/flutter/lib/fix_data.yaml:
```diff
@@ -158,6 +191,39 @@ transforms:
       - kind: 'removeParameter'
         name: 'primaryColorBrightness'
 
+  # Changes made in TBD
+  - title: "Remove 'toggleableActiveColor'"
+    date: 2021-11-11
+    element:
+      uris: [ 'material.dart' ]
+      method: 'copyWith'
+      inClass: 'ThemeData'
+    changes:
+      - kind: 'removeParameter'
+        name: 'toggleableActiveColor'
+
+  # Changes made in TBD
+  - title: "Remove 'toggleableActiveColor'"
+    date: 2021-11-11
+    element:
+      uris: [ 'material.dart' ]
+      constructor: 'raw'
+      inClass: 'ThemeData'
+    changes:
+      - kind: 'removeParameter'
+        name: 'toggleableActiveColor'
+
+  # Changes made in TBD
+  - title: "Remove 'toggleableActiveColor'"
+    date: 2021-11-11
+    element:
+      uris: [ 'material.dart' ]
+      constructor: ''
+      inClass: 'ThemeData'
+    changes:
+      - kind: 'removeParameter'
+        name: 'toggleableActiveColor'
+
```
```suggestion
```

guidezpl commented on packages/flutter/lib/src/material/theme_data.dart:
```diff
@@ -385,6 +384,13 @@ class ThemeData with Diagnosticable {
+    @Deprecated(
+      'No longer used by the framework, please remove any reference to it. '
+      'For more information, consult the migration guide at '
+      'https://flutter.dev/docs/release/breaking-changes/toggleable-active-color#migration-guide. '
```
broken, has it yet to be published to the website?

TahaTesser replied:
Not yet https://github.com/flutter/website/pull/6626

guidezpl reviewed (approved):
I think the analysis error may be due to a missing period after the version

TahaTesser committed:
fix fix_data.yml

skia-gold commented:
Gold has detected about 2 new digest(s) on patchset 16.
View them at https://flutter-gold.skia.org/cl/github/95870

TahaTesser committed:
Update conflicts

TahaTesser committed:
Update deprecation messsage for beta version

TahaTesser committed:
Add missing period.

skia-gold commented:
Gold has detected about 4 new digest(s) on patchset 19.
View them at https://flutter-gold.skia.org/cl/github/95870

skia-gold commented:
Gold has detected about 5 new digest(s) on patchset 20.
View them at https://flutter-gold.skia.org/cl/github/95870

TahaTesser committed:
Remove deprecated toggleableActiveColor from dev/integration_tests/flutter_gallery

skia-gold commented:
Gold has detected about 5 new digest(s) on patchset 21.
View them at https://flutter-gold.skia.org/cl/github/95870

TahaTesser committed:
fix radio doc reference

skia-gold commented:
Gold has detected about 2 new digest(s) on patchset 22.
View them at https://flutter-gold.skia.org/cl/github/95870

TahaTesser commented:
@guidezpl 
Here is the update, the deprecation guide is [live](https://flutter.dev/docs/release/breaking-changes/toggleable-active-color#migration-guide.)

Here are the things I did with this PR 
1. Fix doc reference for ColorScheme.
2. Updated the beta version in the deprecation message.
3. Resolve `fix_data.yaml`, `flutter/test_fixes/material.dart`, `flutter/test_fixes/material.dart.expect` conflicts (due to another depracation PR)
4. Updated `dev/integration_tests/flutter_gallery/lib/gallery/themes.dart` as it uses this color and it is deprecated and test was failing

### Todo

After this lands, I will immediately file website PR to update [timeline](https://docs.flutter.dev/release/breaking-changes/toggleable-active-color#timeline)

Please let me know if there is anything else needed to land this, thank you!

guidezpl commented:
I'm working on pre-migrating internal uses so this won't be a breaking change internally, will report back when done

guidezpl commented:
After some investigation, unfortunately, this won't be able to land without some prerequisite work. Many teams have defined a `toggleableActiveColor` which isn't `ColorScheme.secondary` and so that color must be defined via component themes. 

However, `RadioListTile`, `SwitchListTile` and `CheckboxListTile` don't use the relevant component themes in the same way that `Radio`, `Switch`, and `Checkbox` do, when they were introduced in https://github.com/flutter/flutter/pull/70311. So first, the three list tile components will have to be updated to utilize component themes.

TahaTesser commented:
Closing this for https://github.com/flutter/flutter/pull/97972

sfshaza2 commented:
@TahaTesser, is this definitely landed in 3.7? I'd like to update the breaking changes page.

TahaTesser commented:
Hi @sfshaza2 
Yes, I can confirm `toggleableActiveColor` deprecation is in the latest `stable 3.7` release.
